skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+__author__ = 'Anderson Carlos Ferreira da Silva'
+
+from copy import deepcopy
+from skmultiflow.core.base_object import BaseObject
+from skmultiflow.classification.core.driftdetection.base_drift_detector import BaseDriftDetector
+from skmultiflow.classification.trees.hoeffding_tree import *
+from skmultiflow.classification.core.driftdetection.adwin import ADWIN
+from skmultiflow.classification.trees.arf_hoeffding_tree import ARFHoeffdingTree
+
+INSTANCE_WEIGHT = np.array([1.0])
+FEATURE_MODE_M = ''
+FEATURE_MODE_SQRT = 'sqrt'
+FEATURE_MODE_SQRT_INV = 'sqrt_inv'
+FEATURE_MODE_PERCENTAGE = 'percentage'
+
+
+class AdaptiveRandomForest(BaseClassifier):
+    """Adaptive Random Forest (ARF).
+
+        Parameters
+        ----------
+        nb_ensemble: int (default=10)
+            Number of trees oin the ensemble
+        feature_mode: string (default='sqrt')
```

Comment:
The `feature_mode` and `subspace_size` were needed in the Java version, but in python the `feature_mode` and `subspace_size` could be merged into `max_features` using exactly the approach as in RandomForestClassifier from scikit-learn: 
http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html

Reply:
Agree.
- [x] Moved code into own function _set_max_features, providing same options as scikit-learn RandomForestClassifier

Reply:
max_features : int, float, string or None, optional (default="auto")
            / Number of attributes per subset for each node split.
            / If int, then consider ```max_features``` features at each split.
            / If float, then ```max_features``` is a percentage and ```int(max_features * n_features)```
              features are considered at each split.
            / If "auto", then `max_features=sqrt(n_features)`.
            / If "sqrt", then `max_features=sqrt(n_features)` (same as "auto").
            / If "log2", then `max_features=log2(n_features)`.
            / If None, then `max_features=n_features`.